src/main/java/net/sf/jabref/wizard/auximport/AuxFileParser.java

```diff
@@ -75,196 +71,110 @@ public final int getNotResolvedKeysCount() {
+            result.append(Localization.lang("nested_aux_files")).append(' ').append(nestedAuxCount);
         }
         return result.toString();
     }
 
-    /**
-     * parseAuxFile read the Aux file and fill up some intern data structures. Nested aux files (latex \\include)
-     * supported!
+    private static final Pattern CITE_PATTERN = Pattern.compile("\\\\(citation|abx@aux@cite)\\{(.+)\\}");
+    private static final Pattern INPUT_PATTERN = Pattern.compile("\\\\@input\\{(.+)\\}");
+
+    /*
+     * Parses the aux file and extracts all bib keys.
+     * Also supports nested aux files (latex \\include).
+     *
+     * There exists no specification of the aux file.
+     * Every package, class or document can write to the aux file.
+     * The aux file consists of LaTeX macros and is read at the \begin{document} and again at the \end{document}.
      *
-     *     // found at comp.text.tex
-     //  > Can anyone tell be the information held within a .aux file?  Is there a
-     //  > specific format to this file?
-     //
-     // I don't think there is a particular format. Every package, class
-     // or document can write to the aux file. The aux file consists of LaTeX macros
-     // and is read at the \begin{document} and again at the \end{document}.
-     //
-     // It usually contains information about existing labels
-     //  \\newlabel{sec:Intro}{{1}{1}}
-     // and citations
-     //  \citation{hiri:conv:1993}
-     // and macros to write information to other files (like toc, lof or lot files)
-     //  \@writefile{toc}{\contentsline {section}{\numberline
-     // {1}Intro}{1}}
-     // but as I said, there can be a lot more
-
-     // aux file :
-     //
-     // \\citation{x}  x = used reference of bibtex library entry
-     //
-     // \\@input{x}  x = nested aux file
-     //
-     // the \\bibdata{x} directive contains information about the
-     // bibtex library file -> x = name of bib file
-     //
-     // \\bibcite{x}{y}
-     //   x is a label for an item and y is the index in bibliography
-     * @param filename String : Path to LatexAuxFile
-     * @return boolean, true = no error occurs
+     * BibTeX citation: \citation{x,y,z}
+     * Biblatex citation: \abx@aux@cite{x,y,z}
+     * Nested aux files: \@input{x}
      */
-    private boolean parseAuxFile(String filename) {
-        // regular expressions
-        Matcher matcher;
-
-        // file list, used for nested aux files
-        List<String> fileList = new ArrayList<>(5);
-        fileList.add(filename);
-
-        // get the file path
-        File dummy = new File(filename);
-        String path = dummy.getParent();
-        if (path == null) {
-            path = "";
-        } else {
-            path = path + File.separator;
-        }
-
-        nestedAuxCounter = -1; // count only the nested reads
+    private void parseAuxFile(String filename) {
+        // nested aux files
+        List<String> fileList = Arrays.asList(filename);
```

Comment:
asList(T... a)
    Returns a fixed-size list backed by the specified array.

So it won't work to add file names later on.